uefi/src/proto/device_path/mod.rs

```diff
@@ -266,6 +266,22 @@ impl PartialEq for DevicePathNode {
     }
 }
 
+impl<'a> TryFrom<&[u8]> for &'a DevicePathNode {
+    type Error = ByteConversionError;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        if mem::size_of::<DevicePathHeader>() <= bytes.len() {
+            let dp = <&DevicePathHeader>::try_from(bytes)?;
+            if dp.length as usize <= bytes.len() {
```

Comment:
nit: `usize::from(dp.length)` instead of `as` cast

Reply:
is this because it's more clear to use `as` only when you want it's extra functionality? (e.g. lossy conversion)

Reply:
More clear, and less bug prone. `From` impls are infallible and non-lossy, whereas `as` casts are quite permissive. An incorrect `from` will likely produce a compile error, whereas an incorrect `as` cast might well compile successfully.